src/mbgl/style/paint_property_binder.hpp

```diff
@@ -76,24 +77,24 @@ std::array<float, N*2> zoomInterpolatedAttributeValue(const std::array<float, N>
    `glVertexAttrib*` was unnacceptably slow. Additionally, in GL Native we have
    implemented binary shader caching, which works better if the shaders are constant.
 */
-template <class T, class A>
+template <class P, class T, class A, class Statistics>
```

Comment:
Specializing on unique `P` and `Statistics` increases the binary size significantly:

```
~/Development/mapbox-gl-native $ ../bloaty/bloaty -d symbols build/macos/Release/Mapbox.framework/Versions/Current/Mapbox -- build-before/macos/Release/Mapbox.framework/Versions/Current/Mapbox 
     VM SIZE                                                                                        FILE SIZE
 ++++++++++++++ GROWING                                                                          ++++++++++++++
   +95%  +111Ki [Other]                                                                           +111Ki   +95%
  +2.0% +9.62Ki [None]                                                                           +8.23Ki  +1.7%
  [NEW] +1.06Ki mbgl::FeatureIndex::query(std::__1::unordered_map<std::__1::basic_string<char, s +1.06Ki  [NEW]
  [NEW]    +816 mbgl::style::LineLayer::Impl::queryIntersectsFeature(mbgl::GeometryCoordinates c    +816  [NEW]
  [NEW]    +632 mbgl::style::CompositeFunctionPaintPropertyBinder<mbgl::style::FillOutlineColor,    +632  [NEW]
  [NEW]    +620 mbgl::style::CompositeFunctionPaintPropertyBinder<mbgl::style::CircleColor, mbgl    +620  [NEW]
  [NEW]    +620 mbgl::style::CompositeFunctionPaintPropertyBinder<mbgl::style::IconHaloColor, mb    +620  [NEW]
  [NEW]    +620 mbgl::style::CompositeFunctionPaintPropertyBinder<mbgl::style::TextHaloColor, mb    +620  [NEW]
  [NEW]    +620 mbgl::style::CompositeFunctionPaintPropertyBinder<mbgl::style::CircleStrokeColor    +620  [NEW]
  [NEW]    +620 mbgl::style::CompositeFunctionPaintPropertyBinder<mbgl::style::FillColor, mbgl::    +620  [NEW]
  [NEW]    +620 mbgl::style::CompositeFunctionPaintPropertyBinder<mbgl::style::IconColor, mbgl::    +620  [NEW]
  [NEW]    +620 mbgl::style::CompositeFunctionPaintPropertyBinder<mbgl::style::LineColor, mbgl::    +620  [NEW]
  [NEW]    +620 mbgl::style::CompositeFunctionPaintPropertyBinder<mbgl::style::TextColor, mbgl::    +620  [NEW]
  [NEW]    +464 GCC_except_table681                                                                 +464  [NEW]
  [NEW]    +434 mbgl::LineBucket::getQueryRadius(mbgl::style::Layer const&) const                   +434  [NEW]
   +46%    +432 GCC_except_table37                                                                  +432   +46%
  [NEW]    +426 mbgl::style::CircleLayer::Impl::queryIntersectsFeature(mbgl::GeometryCoordinates    +426  [NEW]
  [NEW]    +384 GCC_except_table717                                                                 +384  [NEW]
  [NEW]    +352 mbgl::style::CompositeFunctionPaintPropertyBinder<mbgl::style::CircleBlur, float    +352  [NEW]
  [NEW]    +352 mbgl::style::CompositeFunctionPaintPropertyBinder<mbgl::style::LineOffset, float    +352  [NEW]
  [NEW]    +352 mbgl::style::CompositeFunctionPaintPropertyBinder<mbgl::style::FillOpacity, floa    +352  [NEW]

 -------------- SHRINKING                                                                        --------------
 -12.8% -13.2Ki [Other]                                                                          -13.2Ki -12.8%
  [DEL]    -642 mbgl::FeatureIndex::query(std::__1::unordered_map<std::__1::basic_string<char, s    -642  [DEL]
  [DEL]    -630 mbgl::style::LineLayer::Impl::queryIntersectsGeometry(mbgl::GeometryCoordinates     -630  [DEL]
  [DEL]    -624 mbgl::style::CompositeFunctionPaintPropertyBinder<mbgl::Color, mbgl::gl::Attribu    -624  [DEL]
 -65.3%    -384 GCC_except_table718                                                                 -384 -65.3%
  [DEL]    -358 mbgl::style::CircleLayer::Impl::queryIntersectsGeometry(mbgl::GeometryCoordinate    -358  [DEL]
  [DEL]    -336 mbgl::style::CompositeFunctionPaintPropertyBinder<float, mbgl::gl::Attribute<flo    -336  [DEL]
  [DEL]    -302 mbgl::style::FillLayer::Impl::queryIntersectsGeometry(mbgl::GeometryCoordinates     -302  [DEL]
  [DEL]    -292 mbgl::style::CompositeFunctionPaintPropertyBinder<mbgl::Color, mbgl::gl::Attribu    -292  [DEL]
  [DEL]    -286 mbgl::style::PaintPropertyBinder<mbgl::Color, mbgl::gl::Attribute<float, 2ul> >:    -286  [DEL]
 -86.4%    -280 GCC_except_table401                                                                 -280 -86.4%
  [DEL]    -280 mbgl::style::CompositeFunctionPaintPropertyBinder<float, mbgl::gl::Attribute<flo    -280  [DEL]
 -52.4%    -260 GCC_except_table682                                                                 -260 -52.4%
  [DEL]    -260 mbgl::style::PaintPropertyBinder<float, mbgl::gl::Attribute<float, 1ul> >::creat    -260  [DEL]
  [DEL]    -248 mbgl::style::PaintPropertyBinder<mbgl::Color, mbgl::gl::Attribute<float, 2ul> >:    -248  [DEL]
 -40.1%    -236 GCC_except_table180                                                                 -236 -40.1%
  [DEL]    -228 mbgl::style::PaintPropertyBinder<float, mbgl::gl::Attribute<float, 1ul> >::creat    -228  [DEL]
  [DEL]    -224 mbgl::style::SourceFunctionPaintPropertyBinder<mbgl::Color, mbgl::gl::Attribute<    -224  [DEL]
 -50.5%    -220 GCC_except_table491                                                                 -220 -50.5%
  [DEL]    -204 GCC_except_table683                                                                 -204  [DEL]
  [DEL]    -200 mbgl::style::SourceFunctionPaintPropertyBinder<mbgl::Color, mbgl::gl::Attribute<    -200  [DEL]

  +3.3%  +112Ki TOTAL                                                                             +110Ki  +3.2%
```

Can we find a way to specialize only on the underlying type (e.g. `float`)?